### DISCUSSION TITLE
REST API: add `frontpage_text` to events

### DISCUSSION THREAD

**Discussion by relikd** [pretix/pretix#2969] [Ideas]
Hi, I would like to retrieve the event description (`frontpage_text`).
This is currently possible for `/events/<slug>/subevents/` but not for `/events`.

**Comment by raphaelm** [comment]
Hi, this is due to the fact that for events, it's implemented as a setting. You can access it through
https://docs.pretix.eu/en/latest/api/resources/events.html#get--api-v1-organizers-(organizer)-events-(event)-settings-

**Reply by relikd** [reply]
Which requires the “Can change event settings” permission. It seems a bit too much to allow a write permission only to retrieve information that is publicly available. I just want to retrieve the title, description, date, and url for all upcoming events. Is there maybe another API that I can use?

**Reply by raphaelm** [reply]
Ah, meh, that's bad. I wonder why we do require that permission, maybe because settings may contain some sensitive data and we have no explicit "can read settings permission". I'm afraid there won't be a trivial solution and there is no other API :/

**Reply by relikd** [reply]
Would it be possible to include the text in `/events` or `/events/<slug>/`? Or have a new endpoint? E.g., an endpoint to retrieve the information that is shown to someone visiting an event page.

**Reply by raphaelm** [reply]
It would be possible of course, but not trivial.  A separate endpoint seams to clutter the API quite a bit. Integrating it into the list of events would be possible but requires quite some clever code to not be a performance regression.

**Reply by relikd** [reply]
Or another idea, create a subscribable ICAL calendar feed? Would be a nice feature for customers too (e.g., always up-to-date events feed).

**Reply by raphaelm** [reply]
An iCal feed exists, but also does not contain the front-page text, as it is jot the front page. We currently don't have a more general "description" field.

**Comment by carl-egge** [comment]
Hi! I just come by this discussion because I wanted to achieve the same. It would be great if the event description (`frontpage_text`) would be accessible through the `/events` endpoint from the API.

Do you really think it would add that much of an overhead? Maybe it can be encoded in order to safe bytes.

Is there any update on this?

**Reply by raphaelm** [reply]
The overhead is not in its size, but in that it is stored in a different place. Also, I believe it's just wrong. frontpage_text is *not* a description. It's a frontpage text. It should only be used for rendering the front page. There currently is no description field, but it would be a valid feature request to add one.
